openslides_backend/action/actions/committee/import_meeting.py

```diff
@@ -0,0 +1,241 @@
+            try:
+                self.check_permissions(instance)
+            except MissingPermission as e:
+                msg = f"You are not allowed to perform action {self.name}."
+                e.message = msg + " " + e.message
+                raise e
+            self.index += 1
+        self.index = -1
+
+        instances = self.get_updated_instances(action_data)
+        for instance in instances:
+            instance = self.base_update_instance(instance)
+            write_request = self.create_write_requests(instance)
+            self.write_requests.extend(write_request)
+
+        final_write_request = self.process_write_requests()
+        return (final_write_request, None)
+
+    def update_instance(self, instance: Dict[str, Any]) -> Dict[str, Any]:
+        meeting_json = instance["meeting_json"]
+
+        # checks if the meeting_json is correct
+        if not len(meeting_json.get("meeting", [])) == 1:
+            raise ActionException("Need exact one meeting in meeting collection.")
+        shall_be_empty = ("organization", "organization_tag", "committee", "resource")
+
+        for collection in shall_be_empty:
+            if meeting_json.get(collection):
+                raise ActionException(f"{collection} must be empty.")
+
+        for user in meeting_json.get("user", []):
+            if not user["password"] == "":
+                raise ActionException("User password must be an empty string.")
+
+        self.check_usernames_and_generate_new_ones(meeting_json)
+        self.update_committee_id(instance)
+        self.generate_random_passwords(meeting_json)
+        self.set_enable_anonymous(meeting_json)
+        self.set_imported_at(meeting_json)
+
+        # replace ids in the meeting_json
+        replace_map = self.create_replace_map(meeting_json)
+        self.replace_fields(instance, replace_map)
+        return instance
+
+    def check_usernames_and_generate_new_ones(self, json_data: Dict[str, Any]) -> None:
+        used_usernames = set()
+        for entry in json_data["user"]:
+            username_unique = False
+            template_username = entry["username"].rstrip("0123456789")
+            count = 1
+            while not username_unique:
+                if entry["username"] in used_usernames:
+                    entry["username"] = template_username + str(count)
+                    count += 1
+                    continue
+                result = self.datastore.filter(
+                    Collection("user"),
+                    FilterOperator("username", "=", entry["username"]),
+                    ["id"],
+                )
+                if result:
+                    entry["username"] = template_username + str(count)
+                    count += 1
+                    continue
+                username_unique = True
+            used_usernames.add(entry["username"])
+
+    def update_committee_id(self, instance: Dict[str, Any]) -> None:
```

Comment:
all these small 1-2 line function can be done in one meeting-altering function.

Reply:
Merge the small set functions into one function.